git/cmd.py

```diff
@@ -305,6 +306,173 @@ def __setstate__(self, d: Dict[str, Any]) -> None:
     the top level ``__init__``.
     """
 
+    _bash_exec_env_var = "GIT_PYTHON_BASH_EXECUTABLE"
+
+    bash_exec_name = "bash"
+    """Default bash command that should work on Linux, Windows, and other systems."""
+
+    GIT_PYTHON_BASH_EXECUTABLE = None
+    """Provide the full path to the bash executable. Otherwise it assumes bash is in the path.
+
+    Note that the bash executable is actually found during the refresh step in
+    the top level ``__init__``.
+    """
+
+    @classmethod
+    def _get_default_bash_path(cls):
```

Comment:
Isn't there a type annotation missing for the return type?

Reply:
I think static type checkers since about Python 3.8 generally no longer require return type annotations if they can be trivially inferred, so by habit I left it off.  I will add one for consistency's sake with everything else here :-)

Reply:
This comment is only in case of interest; it does not unresolve the thread, and it can be ignored.

Although static type checkers have become very capable, which is helpful for things like editor/IDE integration, my understanding is that the *meaning* of a function with no type annotations has remained the same. That is, a function with [no type annotations](https://github.com/gitpython-developers/GitPython/pull/1755#issuecomment-1837419098) still has `Any` as the type of all its parameters and of its return type, and the function is "dynamically typed" (which in practice means that its implementation should not be checked for type errors unless a tool is explicitly told to do so, though some tools have different defaults).

This, or at least `mypy`'s interpretation, can be verified on the [mypy Playground](https://mypy-play.net/?mypy=latest&python=3.12) (which also verifies that the type of a `self` or `cls` argument need not be annotated so long as another annotation is present, i.e., the change that has been made here, though it is valuable for more than just stylistic consistency, is fully sufficient).

Untyped:

```python
class C:
    @classmethod
    def create(cls):
        reveal_type(cls)
        print("x" + 3)
        return cls()

reveal_type(C.create())
```

```text
main.py:4: note: Revealed type is "Any"
main.py:4: note: 'reveal_type' always outputs 'Any' in unchecked functions
main.py:8: note: Revealed type is "Any"
Success: no issues found in 1 source file
```

Typed:

```python
from typing import Self

class C:
    @classmethod
    def create(cls) -> Self:
        reveal_type(cls)
        print("x" + 3)
        return cls()

reveal_type(C.create())
```

```text
main.py:6: note: Revealed type is "Type[Self`0]"
main.py:7: error: Unsupported operand types for + ("str" and "int")  [operator]
main.py:10: note: Revealed type is "__main__.C"
Found 1 error in 1 file (checked 1 source file)
```